src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java

```diff
@@ -147,13 +160,25 @@ public void handle(@Nonnull MutableGlobalContext ctx, @Nonnull Class<?> clazz) t
         }
 
         AutoTransform autoTransform = clazz.getAnnotation(AutoTransform.class);
+        Class<?>[] allTargets = mergeAllTargets(autoTransform);
         Class<?>[] targets = autoTransform.targets();
         for (Class<?> target : targets) {
-            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target), Transformer.class);
+            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target, allTargets), Transformer.class);
+        }
+        Class<?> target = autoTransform.target();
+        if (!target.equals(Object.class)) {
+            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target, allTargets), Transformer.class);
         }
+    }
+
+    private Class<?>[] mergeAllTargets(AutoTransform autoTransform) {
         Class<?> target = autoTransform.target();
         if (!target.equals(Object.class)) {
-            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target), Transformer.class);
+            Class<?>[] targets = autoTransform.targets();
+            Class<?>[] mergedTargets = Arrays.copyOf(targets, targets.length + 1);
```

Comment:
Would this add double entries in the target list?

Reply:
Yes, the array could contain duplicates if the AutoTransform annotation defines them. Those will later be catched at sirius.kernel.di.transformers.Composable#attach(java.lang.Class<? extends T>, T) where the last caller wins. With sane usage, there should be no problem.